components/storage/blockdevice/COMPONENT_SPIF/SPIFBlockDevice.cpp

```diff
@@ -339,12 +339,22 @@ int SPIFBlockDevice::erase(bd_addr_t addr, bd_size_t in_size)
     int size = (int)in_size;
     bool erase_failed = false;
     int status = SPIF_BD_ERROR_OK;
-   // Find region of erased address
+    // Find region of erased address
     int region = _utils_find_addr_region(addr);
     // Erase Types of selected region
     uint8_t bitfield = _region_erase_types_bitfield[region];
 
-    tr_error("DEBUG: erase - addr: %llu, in_size: %llu", addr, in_size);
+    tr_info("DEBUG: erase - addr: %llu, in_size: %llu", addr, in_size);
+
+    if ((addr + in_size) > _device_size_bytes) {
+        tr_error("ERROR: erase exceeds flash device size");
+        return SPIF_BD_ERROR_INVALID_ERASE_PARAMS;
+    }
+
+    if ( ((addr % get_erase_size(addr)) != 0 ) ||  (((addr + in_size) % get_erase_size(addr + in_size - 1)) != 0 ) ) {
```

Comment:
Won't the function `is_valid_erase` work here?

Reply:
is_valid_erase wont work because it relies on a single erase-sector size.
This supports variable erase-sector size according to address.

Reply:
Thanks. This means that it has a bug, which should be fixed (but not in this PR of course).